client/packages/programs/src/JsonForms/common/components/Array/common.tsx

```diff
@@ -214,6 +217,7 @@ export const ArrayCommonComponent = (props: ArrayControlCustomProps) => {
               sx={{
                 mt: '0 !important',
                 mb: index === data.length - 1 ? '20px !important' : 1,
+                border: isError ? '1px solid red' : '',
```

Comment:
Visually, it's not great, but anything more would require a bit of UI design work I think.